extensions/anonaddy/src/api/recipients.ts

```diff
@@ -0,0 +1,24 @@
+import { withCache } from "@raycast/utils";
+
+import fetch from "./fetch";
+
+import type { Paginated, Recipient } from "./types";
+
+// @see https://app.addy.io/docs/#recipients-GETapi-v1-recipients
+const getAll = withCache(
+  async () => {
+    const response = await fetch("recipients?filter[verified]=true");
+
+    if (response.status !== 200) {
+      throw new Error(`Failed to fetch recipients: ${response.status}`);
+    }
+
+    const body = (await response.json()) as Paginated<Recipient>;
+
+    return body.data;
+  },
+  // Cache for 5 minutes
+  { maxAge: 5 * 60 * 1000 },
```

Comment:
This will also cache bad (4xx) requests for 5 mins. This confused me when I first used my API key, and it kept returning an invalid credential error.

What do you think?

Reply:
@LitoMore How so? It should have returned a 401 status code and thrown an error.

Reply:
I think it might be an issue related to the API key provider. I was using a brand new API key created seconds ago, and I used it in the Raycast extension immediately.

This might be a super-edged use case.